packages/datetime/src/timePicker.tsx

```diff
@@ -207,6 +230,41 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
         );
     }
 
+    private maybeRenderAmPm() {
+        if (!this.props.useAmPm) {
+            return null;
+        }
+
+        const onChange = (amPm: string) => {
```

Comment:
pull this to a member function: `private handleAmPmChange`.
currently, it gets redefined every render and breaks any pureness as it's always a new instance.